toolchain/check/handle_binding_pattern.cpp

```diff
@@ -44,6 +44,12 @@ auto HandleBindingPattern(Context& context, Parse::NodeId parse_node) -> bool {
   // Every other kind of pattern binding has a name.
   auto [name_node, name_id] = context.node_stack().PopNameWithParseNode();
 
+  auto make_bind_name = [name_node = name_node, name_id = name_id](
+                            SemIR::TypeId type_id,
+                            SemIR::InstId value_id) -> SemIR::Inst {
+    return SemIR::BindName{name_node, type_id, name_id, value_id};
+  };
```

Comment:
Personal preference, but I think this lamba is increasing code/complexity rather than reducing it. If there's any difference beyond `make_bind_name(type_id, value_id)` versus `SemIR::BindName{name_node, type_id, name_id, value_id}`, I'm missing it. After some reading, I think the only reason `name_node` and `name_id` are taken as copies is the C++20 issue, although is that also in conflict with `[&]`? The lambda requires 5 LOC in order to reduce ~2 LOC, also adding more control flow.

If you keep this, please add a comment at least explaining why the lambda versus direct construction.

Reply:
Added a comment; my next PR is going to add more logic to this lambda, so I'd rather not remove it then re-add it :)